src/sentry/runner/commands/configoptions.py

```diff
@@ -0,0 +1,267 @@
+                else:
+                    click.echo(f"Failed to deleted: {key} (dry run:{dryrun})")
+
+        kv_generator = create_key_value_generator(data, "\n", ": ")
+        for line in kv_generator:
+            click.echo(f"line: {line} {type(line)}")
+            key = line[0]
+            val = ""
+            click.echo(f"key: {key} {type(key)}")
+            if len(line) > 1:
+                val = line[1]
+            click.echo(f"val: {val} {type(val)}")
+            click.echo(f"dryrun: {dryrun} {type(dryrun)}")
+            success = _set(key, val, dryrun)
+            if success:
+                click.echo(f"Successfully updated: {key} = {val} (dry run:{dryrun})")
+            else:
+                click.echo(f"Failed to update: {key} = {val} (dry run:{dryrun})")
+
+
+@configoptions.command()
+@click.argument("key", required=True)
+@click.option(
+    "--dryrun",
+    is_flag=True,
+    default=False,
+    required=False,
+    help="Output exactly what changes would be made and in which order.",
+)
+@configuration
+def get(key: str, dryrun: bool = False) -> str:
+    "Get a configuration option."
+    return _get(key, dryrun)
+
+
+def _get(key: str, dryrun: bool = False) -> str:
+    from sentry.options import default_manager as manager
+    from sentry.options.manager import UnknownOption
+
+    try:
+        opt = manager.lookup_key(key)
+        click.echo(f"Fetched Key: {opt.name} ({opt.type}) = {manager.get(opt.name)}")
+        return opt
+    except UnknownOption:
+        click.echo("unknown option: %s" % key)
+        return ""
+
+
+@configoptions.command()
+@click.argument("key", required=True)
+@click.argument("val", required=True)
+@click.option(
+    "--dryrun",
+    is_flag=True,
+    default=False,
+    required=False,
+    help="Output exactly what changes would be made and in which order.",
+)
+@configuration
+def set(key: str, val: str, dryrun: bool = False) -> bool:
+    return _set(key, val, dryrun)
+
+
+def _set(key: str, val: str, dryrun: bool = False) -> bool:
+    from sentry import options
+    from sentry.options.manager import UnknownOption
+
+    success = False
+    if not can_change(key):
+        return success
+
+    try:
+        opt = options.lookup_key(key)
+
+    except UnknownOption:
+        click.echo(f"Unknown Option in set: {key}")
+        if not dryrun:
+            try:
+                options.register(key)
+                options.set(key, val)
+                success = True
+                click.echo(f"Registered a new key: {key} = {val}")
+            except Exception:
+                click.echo(f"Failed to register option: {key} = {val}")
+
+    else:
+        if not dryrun:
+            try:
+                options.set(key, val)
+                success = True
+                click.echo(f"Updated key: {opt.name} ({opt.type}) = {val}")
+            except Exception:
+                click.echo(f"Failed to update key: {opt.name} ({opt.type}) = {val}")
+
+    return success
```

Comment:
Python has exceptions for this, please do not return opaque status code.
Exceptions are structured, they can provide a lot of context to the caller and have a language native way to be caught and managed. 
By returning a status code you are asking the caller to deviate from python practices to manage error and build something ad hoc.

So please do not log the errors here, raise an exception instead. Then the entrypoint (the @command() methods) can decide how to present the result to the user (errors, stacktrace, etc.)